### PR TITLE
chore(travis): Add travis build and coverage reporting

### PR DESCRIPTION
**Pull request by kentcdodds**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+before_install:
+  - npm i -g npm@^3.0.0
+before_script:
+  - npm prune
+script:
+  - npm run validate
+after_success:
+  - 'curl -Lo travis_after_all.py https://git.io/travis_after_all'
+  - python travis_after_all.py
+  - 'export $(cat .to_export_back) &> /dev/null'
```
Comment (sarbbottam) [inline .travis.yml:20]: Please excuse my ignorance, what does `python travis_after_all.py` do, and who uses the `env var` exported by `'export $(cat .to_export_back) &> /dev/null'`. Thanks!

**Comment by kentcdodds** [comment]
Sweet. Green build and 100% code coverage!

party-excited-birthday-YTbZzCkRQCEJa

**Comment by kentcdodds** [comment]
Someone wanna review and merge this when you get the chance? :-)

**Comment by kentcdodds** [comment]
Again, holding off to do the `semantic-release` stuff until the API stabilizes and comes out of beta.

File: .travis.yml (hunk 1 above)
Comment (kentcdodds) [inline .travis.yml:20]: Honestly, I copy/pasted this from what `semantic-release` generates when you setup a new repo :-) I have no idea what this is for :-)

Comment (kentcdodds) [inline .travis.yml:20]: But let's keep it I say. That way when we add `semantic-release` we can just add it simply

**Comment by sarbbottam** [comment]
Looks good to be merged!! :tada:

**Comment by kentcdodds** [comment]
@sarbbottam feel free to merge it :-)

**Comment by kentcdodds** [comment]
Looks like @sarbbottam is [out](https://twitter.com/sarbbottam/status/710888319840727042) :-) anyone else can merge this as well if you want.

**Comment by nyrosmith** [comment]
@kentcdodds just for you!
BTW very good vibes here. Lots of progress

**Pull request merged**